controller/volume_rebuilding_controller.go

```diff
@@ -192,12 +279,130 @@ func (vbc *VolumeRebuildingController) reconcile(volName string) (err error) {
 	}()
 
 	rebuildingAttachmentTicketID := longhorn.GetAttachmentTicketID(longhorn.AttacherTypeVolumeRebuildingController, volName)
+	defer func() {
+		if err != nil {
+			delete(va.Spec.AttachmentTickets, rebuildingAttachmentTicketID)
+			return
+		}
+	}()
+	if !vol.DeletionTimestamp.IsZero() {
+		vbc.logger.Infof("Volume %v is deleting, skip offline rebuilding", volName)
+		delete(va.Spec.AttachmentTickets, rebuildingAttachmentTicketID)
+		return nil
+	}
+	if vol.Status.Robustness == longhorn.VolumeRobustnessFaulted {
+		vbc.logger.Warnf("Volume %v is faulted, skip offline rebuilding", volName)
+		delete(va.Spec.AttachmentTickets, rebuildingAttachmentTicketID)
+		return nil
+	}
 
-	delete(va.Spec.AttachmentTickets, rebuildingAttachmentTicketID)
+	isOfflineRebuildEnabled, err := vbc.isVolumeOfflineRebuildEnabled(vol.Spec.OfflineRebuilding)
+	if err != nil {
+		return err
```

Comment:
Should delete the VA ticket if err != nil. Right?

Reply:
Yes.